buttercup.el

```diff
@@ -730,26 +707,32 @@ See also `buttercup-define-matcher'."
   "Return the number of failed specs in all suites in SUITE-LIST."
   (buttercup-suites-total-specs-status suite-list 'failed))
 
+(defun buttercup--specs (spec-or-suite-list)
+  "Return a flat list of all specs in SPEC-OR-SUITE-LIST."
+  (let (specs)
+    (dolist (spec-or-suite spec-or-suite-list specs)
+      (if (buttercup-spec-p spec-or-suite)
+          (setq specs (append specs (list spec-or-suite)))
+        (setq specs (append specs (buttercup--specs
+                                   (buttercup-suite-children spec-or-suite))))))))
```

Comment:
Not that it matters much performance-wise, but this is inefficient and highly unidiomatic – could you use the usual push/nreverse or (better, I think) the cl-loop pattern? :-)

Reply:
I think I tried both approaches and the problem is keeping the order.  buttercup--specs-and-suites is equally unappealing.

I'll have a think and see if I can come up with something.

Reply:
Hm. Ah yes, tree recursive stuff. I think I'm actually fine with this then. Thank you!